# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../uclibc/avoid-divdi3.patch
# Copyright (C) 2007 - 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---


WARNING: ugly workaround to prevent linkage with libgcc_s in toolchain stage
Stage 5: /bin/bash: can't resolve symbol '__moddi3' in lib '/bin/bash'.

--- ./libc/stdlib/lldiv.c.orig	2005-08-18 00:49:41.000000000 +0200
+++ ./libc/stdlib/lldiv.c	2005-12-21 23:49:13.000000000 +0100
@@ -28,8 +28,8 @@
 {
     lldiv_t result;
 
-    result.quot = numer / denom;
-    result.rem = numer % denom;
+    result.quot = (long int) numer / (long int) denom;
+    result.rem = (long int) numer % (long int ) denom;
 
     /* The ANSI standard says that |QUOT| <= |NUMER / DENOM|, where
        NUMER / DENOM is to be computed in infinite precision.  In
